internal/cloudapi/openapi.yml

```diff
@@ -0,0 +1,240 @@
+          type: array
+          items:
+            $ref: '#/components/schemas/UploadStatus'
+    UploadStatus:
+      oneOf:
+       - $ref: '#/components/schemas/AWSUploadStatus'
+    AWSUploadStatus:
+      type: object
+      properties:
+        ami_id:
+          type: string
+          example: 'ami-0c830793775595d4b'
+    ComposeRequest:
+      type: object
+      required:
+        - distribution
+        - image_requests
+      properties:
+        distribution:
+          type: string
+          example: 'rhel-8'
+        image_requests:
+          type: array
+          items:
+            $ref: '#/components/schemas/ImageRequest'
+        customizations:
+          $ref: '#/components/schemas/Customizations'
+    ImageRequest:
+      required:
+        - architecture
+        - image_type
+        - repositories
+      properties:
+        architecture:
+          type: string
+          example: 'x86_64'
+        image_type:
+          type: string
+          example: 'ami'
+        repositories:
+          type: array
+          items:
+            $ref: '#/components/schemas/Repository'
+        upload_requests:
+          type: array
+          items:
+            $ref: '#/components/schemas/UploadRequest'
+    Repository:
+      type: object
+      required:
+        - baseurl
+      properties:
+        baseurl:
+          type: string
+          format: url
+          example: 'https://cdn.redhat.com/content/dist/rhel8/8/x86_64/baseos/os/'
+    UploadRequest:
+      type: object
+      required:
+        - type
+        - options
+      properties:
+        type:
+          type: string
+          enum: ['aws']
+        options:
+          oneOf:
+            -  $ref: '#/components/schemas/AWSUploadRequestOptions'
+    AWSUploadRequestOptions:
+      type: object
+      required:
+        - region
+        - s3
+        - ec2
+      properties:
+        region:
+          type: string
+          example: 'eu-west-1'
+        s3:
+          $ref: '#/components/schemas/AWSUploadRequestOptionsS3'
+        ec2:
+          $ref: '#/components/schemas/AWSUploadRequestOptionsEc2'
+    AWSUploadRequestOptionsS3:
+      type: object
+      required:
+        - access_key_id
+        - secret_access_key
+        - bucket
+      properties:
+        access_key_id:
+          type: string
+          example: 'AKIAIOSFODNN7EXAMPLE'
+        secret_access_key:
+          type: string
+          format: password
+          example: 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
+        bucket:
+          type: string
+          example: 'my-bucket'
+    AWSUploadRequestOptionsEc2:
+      type: object
+      required:
+        - access_key_id
+        - secret_access_key
+      properties:
+        access_key_id:
+          type: string
+          example: 'AKIAIOSFODNN7EXAMPLE'
+        secret_access_key:
+          type: string
+          format: password
+          example: 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
+        snapshot_name:
```

Comment:
I added this so we can actually find the snapshot through `describe-images` in the integration test. It will add this as a snapshot Name so we can filter on it.

Ideally we'd have the `awsupload.Register` return the necessary data, have the `AWSUploadStatus` queryable via the cloudapi and then get the information that way.